src/enzyme_ad/jax/Passes/LLVMToAffineAccess.cpp

```diff
@@ -1217,6 +1256,122 @@ convertLLVMToAffineAccess(Operation *op,
     }
   }
 
+  // Raise remaining load/stores to memref.load/memref.store
+  for (const std::pair<Operation *, PtrVal> &p : remainingMemOps) {
+    Operation *memOp = p.first;
+    PtrVal addr = p.second;
+
+    LLVM_DEBUG(llvm::dbgs() << "Building memref access for " << memOp
+                            << " for address " << addr << "\n");
+    auto dl = dataLayoutAnalysis.getAtOrAbove(memOp);
+    if (auto load = dyn_cast<LLVM::LoadOp>(memOp)) {
+      IRRewriter rewriter(load);
+
+      Type ty = load.getType();
+      auto tySize = dl.getTypeSize(ty);
+
+      auto memref0 = mc(addr);
+      Value memref = memref0;
+      auto memrefTy = memref0.getType();
+
+      if (memrefTy.getElementType() != ty) {
+        if (auto p2m = memref.getDefiningOp<enzymexla::Pointer2MemrefOp>())
+          memref = p2m.getOperand();
+        else
+          memref = rewriter.create<enzymexla::Memref2PointerOp>(
+              load.getLoc(), LLVM::LLVMPointerType::get(ty.getContext()),
+              memref);
+        memref = rewriter
+                     .create<enzymexla::Pointer2MemrefOp>(
+                         load.getLoc(),
+                         MemRefType::get(memrefTy.getShape(), ty,
+                                         MemRefLayoutAttrInterface{},
+                                         memrefTy.getMemorySpace()),
+                         memref)
+                     .getResult();
+      }
+
+      SmallVector<Value> indices;
+      // If this is a 1D memref and we have GEP indices, use them
+      if (memrefTy.getRank() == 1 && addr.getDefiningOp<LLVM::GEPOp>()) {
+        auto gepOp = addr.getDefiningOp<LLVM::GEPOp>();
+        if (!gepOp.getIndices().empty()) {
+          auto gepIndex = gepOp.getIndices()[0];
+          Value indexVal;
+          if (auto intAttr = dyn_cast<IntegerAttr>(gepIndex)) {
+            indexVal = rewriter.create<arith::ConstantOp>(
```

Comment:
we don't need to special case constant or not

Reply:
I think there is a need to here - `convertToIndex` needs a `Value` and `getIndices()` returns `array[Union{value,intAttr}]`

Reply:
Ah I forgot Alex and I added the constant attribute not just value, yeah that’s right we need it